src/ripple/overlay/Overlay.h

```diff
@@ -45,6 +47,9 @@ class Overlay
     , public beast::PropertyStream::Source
 {
 protected:
+    using socket_type = boost::asio::ip::tcp::socket;
```

Comment:
If you're using Boost 1.72, note that the `tcp::socket` type uses the type-erasing executor wrapper `net::executor`. You might want to just use a socket with an explicit strand, which would be declared thusly:
```
using socket_type = net::basic_stream_socket<
    net::ip::tcp, net::strand<net::io_context::executor_type> >;
```

Reply:
Am I right that keeping in mind this problem it is recommended to use

`beast::basic_stream< net::ip::tcp, net::strand<net::io_context::executor_type>, unlimited_rate_policy >`

instead of `beast::tcp_stream` as a base for `beast::ssl_stream`?

Reply:
Yes

Reply:
Since we are using Boost 1.70 as current version and 1.71 as planned to migrate to I think that this change can be postponed.